mlrun/projects/project.py

```diff
@@ -3761,6 +3767,79 @@ def _validate_file_path(self, file_path: str, param_name: str):
                 f"<project.spec.get_code_path()>/<{param_name}>)."
             )
 
+    def _resolve_artifact_producer(
+        self,
+        artifact: typing.Union[str, Artifact, DatasetArtifact, ModelArtifact],
+        project_producer_tag=None,
+    ) -> typing.Optional[ArtifactProducer]:
+        """
+        Resolve the artifact producer of the given artifact.
+        If the artifact producer is a project, the artifact is registered with the current project as the producer.
+        If not, the artifact is registered with the original producer. If the artifact is already registered, the
+        artifact is not registered again.
+
+        :param artifact:                The artifact to resolve its producer.
+        :param project_producer_tag:    The tag to use for the project as the producer. If not provided, a tag will be
+        generated for the project.
+        :return:                        A tuple of the resolved producer and the resolved artifact.
+        """
+
+        if not isinstance(artifact, str) and artifact.producer:
+            # if the artifact was imported from a yaml file, the producer can be a dict
+            if isinstance(artifact.spec.producer, ArtifactProducer):
+                producer_dict = artifact.spec.producer.get_meta()
+            else:
+                producer_dict = artifact.spec.producer
+
+            if "project" != producer_dict.get("kind", ""):
+                return ArtifactProducer(
+                    name=producer_dict.get("name", ""),
+                    kind=producer_dict.get("kind", ""),
+                    project=producer_dict.get("project", ""),
+                    tag=producer_dict.get("tag", ""),
+                )
+
+        # the artifact is not registered, register it with the project as the producer
+        project_producer_tag = (
+            project_producer_tag or self._get_hexsha() or str(uuid.uuid4())
+        )
+        return ArtifactProducer(
+            "project",
+            self.metadata.name,
+            self.metadata.name,
```

Comment:
```suggestion
            kind="project",
            project=self.metadata.name,
            name=self.metadata.name,
```

It's better to use named params, and especially in this case where it's not obvious from the actual parameters what each is used for.